packages/api/cms-api/src/dam/files/file-sanitization.interceptor.ts

```diff
@@ -0,0 +1,28 @@
+import { CallHandler, ExecutionContext, Injectable, NestInterceptor } from "@nestjs/common";
+import DOMPurify from "dompurify";
+import fs from "fs";
+import { JSDOM } from "jsdom";
+import { Observable } from "rxjs";
+
+@Injectable()
+export class FileSanitizationInterceptor implements NestInterceptor {
+    private domPurify;
+
+    constructor() {
+        const window = new JSDOM("").window;
+        this.domPurify = DOMPurify(window);
+    }
+
+    intercept(context: ExecutionContext, next: CallHandler<unknown>): Observable<unknown> | Promise<Observable<unknown>> {
+        const ctx = context.switchToHttp();
+        const file = ctx.getRequest().file;
+
+        if (file && file.mimetype === "image/svg+xml") {
+            const fileContent = fs.readFileSync(file.path, "utf-8");
```

Comment:
this is currently blocking. Could it possibly be converted to an async call?

Reply:
That's a valid point. I changed it to be non-blocking